src/App.tsx

```diff
@@ -8,7 +8,69 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { FilterBy } from './types/Filter';
+
+const filteredTodos = (todos: Todo[], query: string, filterBy: FilterBy) => {
+  let newTodos = [...todos];
+
+  if (query) {
+    const newQuery = query.toLowerCase().trim();
+
+    newTodos = newTodos.filter(todo => todo.title.toLowerCase().includes(newQuery));
+  }
+
+  switch (filterBy) {
+    case FilterBy.ACTIVE:
+      newTodos = newTodos.filter(todo => !todo.completed);
+      break;
+    case FilterBy.COMPLETED:
+      newTodos = newTodos.filter(todo => todo.completed);
+      break;
+    case FilterBy.ALL:
+    default:
+      break;
+  }
+
+  return newTodos;
+};
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.ALL);
+
+  const [load, setLoad] = useState(true);
+  const [error, setError] = useState(false);
+
+  const getTodosServer = async () => {
+    try {
+      const arrayTodos = await getTodos();
+
+      setTodos(arrayTodos);
+      setLoad(false);
+      setError(false);
+    } catch {
+      setError(true);
+    }
+  };
+
+  useEffect(() => {
+    getTodosServer();
+  }, []);
+
+  const visibleTodos = filteredTodos(todos, query, filterBy);
+
+  const showTodo = useCallback((todo: Todo) => {
+    setSelectedTodo(todo);
+  }, [selectedTodo]);
```

Comment:
wrong value in the dependency array
you shouldn't recreate this function every time when selectedTodo changes. Empty dependency array or `setSelectedTodo` there would be correct choice

But actually, you don't need this wrapper at all